contrib/docker/docker-compose.yml

```diff
@@ -20,7 +20,7 @@ services:
     image: redis:3.2
     restart: unless-stopped
     ports:
-      - 6379:6379
+      - "127.0.0.1:6379:6379"
```

Comment:
I've never seen this docker-compose notation before.  Why are you declaring the ports in one line and then re-declaring them with the localhost on the next?  This is most unusual.  Same for line 35 below...

Reply:
This mean that you only open the port on the host loop interface. Otherwise it will be open in main network interface and overrides iptable blocking rules.
I added this because first of all we don't need to expose the port at all. And second if we need it for debug it's better to be on loop interface to maintain security in the service. (Take a look at [this](https://fralef.me/docker-and-iptables.html) article)

Reply:
Ah, I see what you're getting at.  I run all of my services behind a reverse proxy so I never expose ports directly anyway.  Regardless, I prefer to expose the port and do all of the container-to-container communication through docker itself and not use the localhost at all.  But I can see how this fix might be needed for some situations. 

I still think we should update the docker-compose to v3 and use 'networks' instead of 'depends-on'.  That would make this more explicit, IMHO.

Reply:
I do agree with you, but I don't know whether it fixes the problem or no.
I think it can wait.